ddev-stop -a and ddev start -a result in a panic if no projects are available

**Describe the bug**

Stop and unlist all projects: `ddev stop --unlist -a`
Then run `ddev stop -a`

```
$ ddev stop -a
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/drud/ddev/cmd/ddev/cmd.glob..func42(0x1d9ac40, 0xc0001ad5d0, 0x0, 0x1)
	/workdir/cmd/ddev/cmd/stop.go:58 +0x473
github.com/spf13/cobra.(*Command).execute(0x1d9ac40, 0xc0001ad5c0, 0x1, 0x1, 0x1d9ac40, 0xc0001ad5c0)
	/workdir/vendor/github.com/spf13/cobra/command.go:846 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x1d99f20, 0x1f5, 0x0, 0x0)
	/workdir/vendor/github.com/spf13/cobra/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/workdir/vendor/github.com/spf13/cobra/command.go:887
github.com/drud/ddev/cmd/ddev/cmd.Execute()
	/workdir/cmd/ddev/cmd/root.go:153 +0x3c
main.main()
	/workdir/cmd/ddev/main.go:21 +0xd9
```

**To Reproduce**
Steps to reproduce the behavior:

1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Version and configuration information (please complete the following information):**

- Host computer OS and Version: [e.g. Windows 10, macOS Catalina]
- Docker Desktop version if on macOS or Windows (from "About Docker Desktop")
- ddev version information (use `ddev version`)
- config.yaml contents for the misbehaving project
- Do you have any custom configuration (nginx, php, mysql) in the .ddev folder? If so, have you tried without them?

**Additional context**
Add any other context about the problem here. Thanks!

--- a/cmd/ddev/cmd/delete.go
+++ b/cmd/ddev/cmd/delete.go
@@ -32,7 +32,9 @@ ddev delete --all`,
 		if err != nil {
 			util.Failed("Failed to get project(s): %v", err)
 		}
-		instrumentationApp = projects[0]
+		if len(projects) > 0 {
+			instrumentationApp = projects[0]
+		}
 
 		// Iterate through the list of projects built above, removing each one.
 		for _, project := range projects {

--- a/cmd/ddev/cmd/pause.go
+++ b/cmd/ddev/cmd/pause.go
@@ -21,7 +21,9 @@ in any directory by running 'ddev pause projectname [projectname ...]' or pause 
 		if err != nil {
 			util.Failed("Unable to get project(s): %v", err)
 		}
-		instrumentationApp = projects[0]
+		if len(projects) > 0 {
+			instrumentationApp = projects[0]
+		}
 
 		for _, project := range projects {
 			if err := ddevapp.CheckForMissingProjectFiles(project); err != nil {

--- a/cmd/ddev/cmd/restart.go
+++ b/cmd/ddev/cmd/restart.go
@@ -28,7 +28,9 @@ ddev restart --all`,
 		if err != nil {
 			util.Failed("Failed to get project(s): %v", err)
 		}
-		instrumentationApp = projects[0]
+		if len(projects) > 0 {
+			instrumentationApp = projects[0]
+		}
 
 		for _, app := range projects {
 

--- a/cmd/ddev/cmd/snapshot.go
+++ b/cmd/ddev/cmd/snapshot.go
@@ -21,7 +21,9 @@ ddev snapshot --all`,
 		if err != nil {
 			util.Failed("Unable to get project(s) %v: %v", args, err)
 		}
-		instrumentationApp = apps[0]
+		if len(apps) > 0 {
+			instrumentationApp = apps[0]
+		}
 
 		for _, app := range apps {
 			if snapshotNameOutput, err := app.Snapshot(snapshotName); err != nil {

--- a/cmd/ddev/cmd/ssh.go
+++ b/cmd/ddev/cmd/ssh.go
@@ -24,7 +24,7 @@ ddev ssh -d /var/www/html`,
 	Args: cobra.MaximumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 		projects, err := getRequestedProjects(args, false)
-		if err != nil {
+		if err != nil || len(projects) == 0 {
 			util.Failed("Failed to ddev ssh: %v", err)
 		}
 		app := projects[0]

--- a/cmd/ddev/cmd/start.go
+++ b/cmd/ddev/cmd/start.go
@@ -40,7 +40,9 @@ ddev start --all`,
 		if err != nil {
 			util.Failed("Failed to get project(s): %v", err)
 		}
-		instrumentationApp = projects[0]
+		if len(projects) > 0 {
+			instrumentationApp = projects[0]
+		}
 
 		for _, project := range projects {
 			if err := ddevapp.CheckForMissingProjectFiles(project); err != nil {

--- a/cmd/ddev/cmd/stop.go
+++ b/cmd/ddev/cmd/stop.go
@@ -55,7 +55,9 @@ ddev stop --remove-data`,
 		if err != nil {
 			util.Failed("Failed to get project(s): %v", err)
 		}
-		instrumentationApp = projects[0]
+		if len(projects) > 0 {
+			instrumentationApp = projects[0]
+		}
 
 		// Iterate through the list of projects built above, removing each one.
 		for _, project := range projects {

